Add xml file for running ND reconstruction using LArSED input.

yangtj207 opened the pull request:
Add a xml file for running ND reconstruction using LArSED input and with LArHierarchyValidation.

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+        <MinHitsForDisplay>0</MinHitsForDisplay>
+    </algorithm>
+
+<!--
```
I would like this xml file to only use the hierarchy tools, so we should remove this `LArNeutrinoEventValidation` code block, even though it is commented out.

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+<algorithm type = "LArHierarchyValidation">
+   <ValidateEvent>true</ValidateEvent>
+   <WriteTree>true</WriteTree>
+   <FileName>EventValidation.root</FileName>
```
Perhaps we should use `EventHierarchy.root` and `EventHierarchy` for the tree name? Then we won't potentially confuse new `hierarchy` output files with previous `validation` ones.

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+<algorithm type = "LArHierarchyValidation">
+   <ValidateMC>true</ValidateMC>
+   <WriteTree>true</WriteTree>
+   <FileName>MCValidation.root</FileName>
```
Similarly, could we use `MCHierarchy.root` and `MCHierarchy` for the tree name?

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+    </algorithm>
+-->
+
+<!--
```
Perhaps add a text comment here such as `enable this for event visualisation` ?

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+    <SingleHitTypeClusteringMode>true</SingleHitTypeClusteringMode>
+
+    <!-- ALGORITHM SETTINGS -->
+    <!--algorithm type = "LArEventReading"/-->
```
This line can be removed (`LArEventReading`) ?

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+        <OutputFile>Validation.root</OutputFile>
+    </algorithm>
+-->
+<!--
```
Should we replace this with another call to `LArHierarchyMonitoring`? Also add a comment text: "enable to see PFOs"

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+    </algorithm>
+
+<!--
+    <algorithm type = "LArVisualMonitoring">
```
Is `LArVisualMonitoring` now replaced by the use of `LArHierarchyMonitoring`? If so, can we remove this?

jback08 commented on settings/PandoraSettings_LArSED.xml:
```diff
@@ -0,0 +1,121 @@
+-->
+
+<!--
+    <algorithm type = "LArVisualParticleMonitoring">
```
Do we have hierarchy equivalents for these visualisation algorithms?

jback08 reviewed (changes requested):
Thanks for this, Tingjun. 

I would like this file to be the one that enables hierarchy tools for both SED and EDepSim data formats, and so this should be renamed to something like `PandoraSettings_LArRecoNDHierarchy.xml`, since the xml file works for any data format.

I have some comments and questions below about the various algorithm calls.

yangtj207 commented:
Hi @jback08 I've addressed most of your comments. I simply removed all commented out blocks since I figured we could add them back if needed in the future.

jback08 commented:
Thanks for the updates, @yangtj207 . However, could you please add back in all of the event display `hierarchy` settings, although leave them commented out, so that the user just has to uncomment them to be able to use the display, i.e. what complete set of xml commands did you have in order to create the plots shown in the DUNE-ND slack channel?

I just wanted to remove all instances of the `validation` settings.

jback08 commented:
OK, I was misunderstanding what was required for the visualisation steps, and this simplifies the xml file. Thanks, Tingjun.
